### PR TITLE
Design document for jdbc driver.

### PR DESCRIPTION
**Pull request by kneeraj**
Have added the details of 'load-balancer' and data aware routing both. some work still left. but please review.

### CODE REVIEW

File: architecture/design/smart-driver.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
+<!-- Output copied to clipboard! -->
+
+
+<h1>Smart JDBC driver for YSQL</h1>
```
Comment (rkarthik007) [inline architecture/design/smart-driver.md:4]: Please use the markdown for Github... for example:

* If you write:
`# Smart JDBC driver`
it renders as:
# Smart JDBC driver

* For subheaders below, if you write:
* `## Motivation`
it renders as:
## Motivation

And use `### XYZ` for the third level heading.

File: architecture/design/smart-driver.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+
+<h2>Motivation</h2>
+
+
+The current Yugabyte JDBC driver is built on the Postgres JDBC driver. It works fine with YugabyteDB as YugabyteDB is wire compatible with PostgreSQL. However, YugabyteDB being a distributed database can use some smarts to not only increase performance but also obviate the need of external load balancers like HAProxy, pgpool etc. Elaborating on the four key motivations.
+
+<h4>Load Balancing</h4>
```
Comment (rkarthik007) [inline architecture/design/smart-driver.md:12]: As discussed, lets make this `Cluster awareness`. See the terminology here: https://github.com/yugabyte/jdbc-yugabytedb

File: architecture/design/smart-driver.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+
+Postgres creates a separate backend child process for every connection. This limits the number of possible processes which can be created on a particular machine owing to resources available. This means that the maximum number of clients which can be created on a YugabyteDB cluster of a limited number of servers is also limited. One can argue about adding more and more servers to the cluster if required. This may be a theoretically valid argument but can never be an economically viable option.
+
+The solution to this obviously seems to be to share a single physical connection for multiple logical connections. However this would entail massive changes at the protocol layer and session management layer. The PostgreSQL wire protocol assumes synchronous execution and sharing would mean asynchronous processing which would need modifications at the protocol layer, state maintenance etc.
+
+<h2>Design </h2>
+
+
+<h3>**Load Balancing**</h3>
```
Comment (rkarthik007) [inline architecture/design/smart-driver.md:37]: As above, just use section headers with `### ...` - no need of `<h>` tags

**Review by rkarthik007 — COMMENTED** [review]
Great doc, left a few comments. We also discussed removing sections on follower reads, and moving some parts like shard-awareness to future work.

**Review by m-iancu — APPROVED** [review]
LGTM!

**Pull request merged**